ligo/haskell/src/Ligo/BaseDAO/Types.hs

```diff
@@ -177,6 +180,21 @@ mkStorageL admin votingPeriod quorumThreshold extra metadata =
     votingPeriodDef = 60 * 60 * 24 * 7  -- 7 days
     quorumThresholdDef = 4
 
+mkStorageL'
+  :: "admin" :! Address
+  -> "votingPeriod" :? Natural
+  -> "quorumThreshold" :? Natural
+  -> "extra" :! ContractExtraL
+  -> "metadataHostAddress" :! Address
+  -> "metadataHostChain" :? TZIP16.ExtChainId
+  -> "metadataKey" :! MText
+  -> StorageL
+mkStorageL' admin votingPeriod quorumThreshold extra hostAddress hostChain key =
+  mkStorageL admin votingPeriod quorumThreshold extra ! #metadata metadata
+  where
+    host = maybe TZIP16.contractHost TZIP16.foreignContractHost (argF #metadataHostChain hostChain) (arg #metadataHostAddress hostAddress)
+    metadata = TZIP16.metadataURI (TZIP16.tezosStorageUri host (arg #metadataKey key))
```

Comment:
Instead of adding more specific function I'd suggest to keep using the existing `metadata` field of `mkStorageL` and `mkFullStorageL`.

To provide the convenience of building it from the 3 parameters a new function that produces a `TZIP16.MetadataMap BigMap` with these two lines would be better suited IMO.

Also, a nitpick: please split the first one to fit into the line length limits.

Reply:
Good suggestion! All done.